Use tasks.named for integrationTestTaskAssigned

rwinch opened the pull request:
Previously Gretty used `tasks."${taskName}"` to access the farm integration
tasks. This caused the tasks to be initialized at configuration time. The
impact is very noticeable in projects that had a lot of projects. A real
world project running into this is Spring Security. Applying
the Gretty plugin in Spring Security's build results in the Gretty
Plugin's Collection callbacks taking about 3.6 seconds of configuration
time.

To better isolate performance of the Gretty Plugin I created
https://github.com/rwinch/gretty-performance-configuration-callback In
this project applying the Gretty plugin results in the Gretty Plugin's
Collection callbacks taking about 5.5 seconds of configuration time.

A link to the scan is found below:
https://scans.gradle.com/s/nsn2j4x3qroa2

With these changes the Collection callbacks is reduced to 0.39 seconds.

A link to the improved scan is found below:
https://scans.gradle.com/s/g6crbfgfeqmfm

Issue gh-141

boris-petrov commented:
@rwinch - thank you for taking the time to improve this. Any ideas why the build is failing?

rwinch commented:
Sorry about that. I had only ran `./gradlew build` and missed running the integration tests that are being ran in travis.

It appears that the issue is related to https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_general

> For example, Project.afterEvaluate() cannot be called when configuring a task registered with the new API. Since afterEvaluate is used to delay configuring a Project, mixing delayed configuration with the new API can cause errors that are hard to diagnose because tasks registered with the new API are not always configured, but an afterEvaluate block may be expected to always execute.

I'm not sure I'm going to have the time to sort this out given how much gretty relies on such hooks.

rwinch commented:
Thanks for the pointer to my mistake around the negation @boris-petrov! I rebased off of master and fixed the logic and force pushed to my branch. For some reason, the updates are not appearing in the PR, but you [can see I've made the changes](https://github.com/gretty-gradle-plugin/gretty/compare/master...rwinch:gh-141-tasks-named) in the branch. I suspect it might not be showing up in the PR due to the [degraded performance of GitHub Pull Requests](https://www.githubstatus.com/incidents/wdhkj70kynwk)?

![image](https://user-images.githubusercontent.com/362503/110985459-180c6100-8332-11eb-9016-20b13c6a4c0e.png)

boris-petrov commented:
@rwinch - thanks! Indeed, it's not visible here. Perhaps you can try force-pushing or something again? I guess GitHub is up again. :)

rwinch commented:
@boris-petrov Thanks for the nudge. I had to make a very small modification to the commit so that a force push was actually made (I modified the commit message very slightly). It seems it is now picked up by this PR.
